pennylane/gradients/gradient_transform.py

```diff
@@ -266,10 +265,10 @@ def _all_zero_grad(tape, shots=Shots(None)):
 
         list_zeros.append(sub_list_zeros)
 
-    if shots.has_partitioned_shots:
+    if tape.shots.has_partitioned_shots:
         if len(tape.measurements) == 1:
-            return [], lambda _: tuple(list_zeros[0] for _ in range(shots.num_copies))
-        return [], lambda _: tuple(tuple(list_zeros) for _ in range(shots.num_copies))
+            return [], lambda _: tuple(list_zeros[0] for _ in range(tape.shots.num_copies))
+        return [], lambda _: tuple(tuple(list_zeros) for _ in range(tape.shots.num_copies))

```

Comment:
Completely optional, but we could leave this code cleaner than we found it and turn the lambas into normal functions.
```suggestion
           
           num_copies = tape.shots.num_copies
           def zero_grad(*_):
               return tuple(list_zeros[0] for _ in range(num_copies)

            return [], zero_jacobian
            
        num_copies = tape.shots.num_copies
        def zero_grad(*_):
           return tuple(list_zeros for _ in range(num_copies)
        return [], zero_grad
```